browser/ui/toolbar/brave_app_menu_model.cc

```diff
@@ -284,18 +250,175 @@ void BraveAppMenuModel::InsertBraveMenuItems() {
+  // download and extensions.
+  absl::optional<size_t> bookmark_item_index =
+      GetIndexOfCommandId(IDC_BOOKMARKS_MENU);
+
+  // If bookmark is not used, we don't need to adjust download item.
+  if (bookmark_item_index.has_value()) {
+    // Place download menu under bookmark.
+    DCHECK(IsCommandIdEnabled(IDC_SHOW_DOWNLOADS));
+    RemoveItemAt(GetIndexOfCommandId(IDC_SHOW_DOWNLOADS).value());
+    InsertItemWithStringIdAt(bookmark_item_index.value(), IDC_SHOW_DOWNLOADS,
+                             IDS_SHOW_DOWNLOADS);
+  }
+
+  if (IsCommandIdEnabled(IDC_MANAGE_EXTENSIONS)) {
+    InsertItemWithStringIdAt(
+        GetIndexOfCommandId(IDC_SHOW_DOWNLOADS).value() + 1,
+        IDC_MANAGE_EXTENSIONS, IDS_SHOW_EXTENSIONS);
+  }
+}
+
+void BraveAppMenuModel::BuildMoreToolsSubMenu() {
+  ui::SimpleMenuModel* more_tools_menu_model =
+      static_cast<ui::SimpleMenuModel*>(
+          GetSubmenuModelAt(GetIndexOfCommandId(IDC_MORE_TOOLS_MENU).value()));
+  DCHECK(more_tools_menu_model);
+
+  size_t next_target_index = 0;
+  bool need_separator = false;
+
+  // Create New Profile
+  if (IsCommandIdEnabled(IDC_ADD_NEW_PROFILE)) {
+    more_tools_menu_model->InsertItemWithStringIdAt(
+        next_target_index++, IDC_ADD_NEW_PROFILE, IDS_ADD_NEW_PROFILE);
+    need_separator = true;
+  }
+
+  // Open Guest Window
+  if (IsCommandIdEnabled(IDC_OPEN_GUEST_PROFILE)) {
+    more_tools_menu_model->InsertItemWithStringIdAt(
+        next_target_index++, IDC_OPEN_GUEST_PROFILE, IDS_OPEN_GUEST_PROFILE);
+    need_separator = true;
+  }
+
+  if (need_separator) {
+    more_tools_menu_model->InsertSeparatorAt(next_target_index++,
+                                             ui::NORMAL_SEPARATOR);
+    need_separator = false;
+  }
+
+#if defined(TOOLKIT_VIEWS)
+  if (sidebar::CanUseSidebar(browser())) {
+    sub_menus_.push_back(std::make_unique<SidebarMenuModel>(browser()));
+    more_tools_menu_model->InsertSubMenuWithStringIdAt(
+        next_target_index++, IDC_SIDEBAR_SHOW_OPTION_MENU,
+        IDS_SIDEBAR_SHOW_OPTION_TITLE, sub_menus_.back().get());
+    more_tools_menu_model->InsertSeparatorAt(next_target_index++,
+                                             ui::NORMAL_SEPARATOR);
   }
 #endif
+
+  if (media_router::MediaRouterEnabled(browser()->profile())) {
+    more_tools_menu_model->InsertItemWithStringIdAt(
+        next_target_index++, IDC_ROUTE_MEDIA, IDS_MEDIA_ROUTER_MENU_ITEM_TITLE);
+    need_separator = true;
+  }
+
+  // Insert sync menu
+  if (IsCommandIdEnabled(IDC_SHOW_BRAVE_SYNC)) {
+    more_tools_menu_model->InsertItemWithStringIdAt(
+        next_target_index++, IDC_SHOW_BRAVE_SYNC, IDS_SHOW_BRAVE_SYNC);
+    need_separator = true;
+  }
+
+  if (need_separator) {
+    more_tools_menu_model->InsertSeparatorAt(next_target_index++,
+                                             ui::NORMAL_SEPARATOR);
+    need_separator = false;
+  }
+
+  if (auto index =
```

Comment:
If the task manager item doesn't exist, should we just insert this as the last item?

Reply:
the code in PR can still skip `IDC_DEV_TOOLS` insertion if `IDC_TASK_MANAGER` is not available.

Reply:
If `IDC_TASK_MANAGER` is not available, we don't remove `IDC_DEV_TOOLS` item - https://github.com/brave/brave-core/pull/19833/files#diff-2283271e14350311184ccfe17b67de5e026532933dd04faea092caa3611f4871R427